### ISSUE TITLE
connect/echo example doesn't seem to work

### ISSUE THREAD

**Issue by dhylands** [tokio-rs/tokio#1568]
## Version

I cloned the latest master ab785bfba774001ce3ba36cc06e88583634eee84

## Platform

ubuntu 19.04

```
Linux dave-work-old 5.0.0-27-generic #28-Ubuntu SMP Tue Aug 20 19:53:07 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```
## Description

Following the comments at the top of exmplaes/echo.rs, in one window I ran:
```
cargo run --example echo
```
and in anorher I ran:
```
cargo run --example connect 127.0.0.1:8080
```
The connection succeeds, but the read in the echo server never completes (or possibly the write on the connection side never completes).

I was using:
```
rustc 1.39.0-nightly (bea0372a1 2019-08-20)
```

**Comment by tmiasko** [comment]
The `connect.rs` writes to `FramedWrite` through `SinkExt::send_all`, that
flushes only once when input stream is exhausted (this looks like a bug
though, or at least suboptimal implementation), and the `FramedWrite`
itself flushes the transport either when buffer is over 8KiB, i.e.,
`BACKPRESSURE_BOUNDARY`. Thus example is not useful for interactive use.

Possible fixes could include:

* Sending each item individually with `SinkExt::send`.
* Switching sink / stream order and using `StreamExt::forward` which inserts
  flushes when stream is pending.
* Fixing `SinkExt::send_all` to insert flushes in similar manner (it looks like
  there is a typo `s/poll_ready/poll_flush/`).

**Comment by dhylands** [comment]
Thanks for the insight. If I enter a bunch of lines and then press Control-D, I do see the lines echoed. Or if I do:
```
cargo run --example connect 127.0.0.1:8080 < some-file.txt
```
then I'll see the contents of some-file.txt echo'd back.

**Comment by LucioFranco** [comment]
@tmiasko I think you're right here at a quick glane, send_all should be the same implementation as forward but on `StreamExt`.

**Comment by carllerche** [comment]
@LucioFranco  What are the actions we need to take to close this issue?

**Comment by LucioFranco** [comment]
We should check if this behavior was fixed with futures 0.3 :)

**Comment by Darksonn** [comment]
I ran the example and it appears to work.
